packages/snap/src/handlers/set-network.ts

```diff
@@ -22,15 +27,29 @@ const coerceParams = (params: SetNetworkParams): bigint => {
 };
 
 /**
- * @description Set the current network using a chain id
+ * @description Set the current network using a rpc url
  * @param params - The network parameters
- * @returns The network chain id
+ * @returns The network rpc url
  */
 export const setNetwork: Handler<SetNetworkParams, SetNetworkResponse> = async (
   params,
 ) => {
-  const network = coerceParams(params);
+  if (!params.network || typeof params.network !== 'string') {
+    throw new Error('Invalid params: network must be a string');
+  }
+
+  const network = params.network;
+
+  const config: IClientConfig = {
+    providers: [{ url: network, type: ProviderType.PUBLIC }],
+  };
+
+  const publicApi = new PublicApiClient(config);
+  const node_status = await publicApi.getNodeStatus();
+
+  await setActiveRPC(network);
+  await setActiveChainId(node_status.chain_id);
 
-  await setActiveChainId(network);
+  console.log('Network set to:', network);
```

Comment:
console.log to remove (idk if its really an isssue or not.... but still)

Reply:
done